### DISCUSSION TITLE
issues with cloudflare

### DISCUSSION THREAD

**Discussion by deanfourie1** [firefly-iii/firefly-iii#8712] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi im trying to deploy this with cloudflare, for remote use.

However, when accessing the webapp on mobile via the cloudflare link, the menus do not work?
I know this may not be directly a firefly related issue, but I would love some help resolving why this could be hapening?

Thanks

### Debug information

Debug information generated at 2024-03-23 09:47:02 for Firefly III version **v6.1.12**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.12 / v2.0.12 / 23 (exp. 23)</td></tr><tr><td>PHP version</td><td>8.3.4 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1037, base <span>#</span>82</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Amsterdam + [BrowserTZ]</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>never (never)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td></td></tr><tr><td>Session start</td><td>2024-03-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-03-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (iPhone; CPU iPhone OS 17_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/123.0.6312.52 Mobile/15E148 Safari/604.1</td></tr></tbody></table>

### Expected behaviour

menus to work as they do when accessing the webpage locally.

### Steps to reproduce

1. Install as per the instruction
2. setup cloudflared tunnel and route the application via http://localhost in cloudflare settings
3. access the url at myapp.mydomain.com and attempt to use the + menu and the main menu, they do nothing.

### Additional info

_No response_no

**Comment by JC5** [comment]
I bet you need to set TRUSTED_PROXIES=* and then it works :)

**Reply by deanfourie1** [reply]
Sorry where do I set this? Is this an environment variable? Im running in docker.

**Reply by JC5** [reply]
Yes.

**Reply by deanfourie1** [reply]
Thanks, does this app support weekly budgeting?

**Reply by deanfourie1** [reply]
Sorry, but setting 

```
# TRUSTED_PROXIES is a useful variable when using Docker and/or a reverse proxy.
# Set it to ** and reverse proxies work just fine.
TRUSTED_PROXIES=**
```

Did not work for me, I also tried with 1 *, still no go.

Any idea?

**Reply by deanfourie1** [reply]
From the browser logs.


`login:1 Refused to load the script 'https://domain.com/cdn-cgi/scripts/7d0fa10a/cloudflare-static/rocket-loader.min.js' because it violates the following Content Security Policy directive: "script-src 'unsafe-eval' 'strict-dynamic' 'nonce-UVt/bZyvPcs/rbWpMTf6Hg=='". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.`

**Reply by JC5** [reply]
Those scripts are added by Cloudflare, not by me. And they wouldn't stop the other scripts from loading. Are there any other errors?

**Comment by deanfourie1** [comment]
FYI - I solved this by creating a custom page rule that disables the rocket-loader in CF dashboard.

Thanks for the help

**Reply by teishahbc** [reply]
Same issue for me and it helps. Thank you！

**Comment by deanfourie1** [comment]
Ok so i've almost got it working, the only thing that still doesnt work is it does not like the anchors?

Everytime I click somewhere that has the #, it just reloads the page, also it does not load any of the transaction pages. Such as a new expense.

**Comment by deanfourie1** [comment]
Preview of the problem.

https://github.com/firefly-iii/firefly-iii/assets/71812804/c564a6ab-a154-4a04-8aae-322056834860

**Reply by JC5** [reply]
What errors do you see in the browser console?

**Comment by fedor-git** [comment]
I’m experiencing an issue with Firefly III, and I’ve enabled debug mode to gather more information. However, most of the logs haven’t provided useful insights into the problem, except for one specific log entry.

The only significant entry in the debug logs is:
```
prod.DEBUG: Error "Illuminate\Session\TokenMismatchException" has no Firefly III treatment, parent will handle.
```

**Comment by sidhantunnithan** [comment]
I am facing this same issue. Disabling rocket loader fixed some issues as OP said, but the new transaction option is still broken behind CF Tunnel. Is there any solution for this as of May 2025?

**Reply by sidhantunnithan** [reply]
I have found the solution for this. Initially, I had added a page rule under **Rules > Page Rules**.

Instead I had to add it under **Rules > Overview > Create Rule > Create Configuration Rule** and add a wildcard on **URI Full** like so : `https://firefly.<domain>.com/*` and turn off **Rocket Loader** for this rule
